pkg/controller/job/util/utils_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestGetFinishedStatus(t *testing.T) {
+	tests := []struct {
+		name            string
+		conditionType   batch.JobConditionType
```

Comment:
Instead of passing `conditionType` and `conditionStatus` it is more flexible to pass `[]batch.JobCondition`. It would let us test cases like no condition, or multiple conditions.